plugins/modules/ec2_vpc_igw.py

```diff
@@ -156,6 +159,31 @@ def get_matching_igw(self, vpc_id):
 
         return igw
 
+    def wait_for_igw(self, vpc_id):
```

Comment:
I'm really not a fan of these home-brew waiters, have you tried adding a custom waiter to module_utils/waiters.py ?  That *should* be more stable than the classic waiters, our waiter code automatically injects retries for the common temporary failures.

Reply:
Yeah, we tried a bunch of different things (see #647). This ones a bit strange and will return it's ready then immediately return null in the next call of describe_igw.

Reply:
This sounds really familiar and I recall doing a batch of similar stability work a while back.

Rather than custom code like this, try using the usual waiters again and then calling describe_igws_with_backoff with `InternetGatewayIds=[gateway_id]` instead of `Filters=...`

Reply:
That is interesting.. I'll update